### DISCUSSION TITLE
Quick Question About graph_iterator_node.py

### DISCUSSION THREAD

**Discussion by RyanGNguyen** [ScrapeGraphAI/Scrapegraph-ai#369] [General]
Hi, I've been experimenting with using this library for a project that I'm working on and was looking through graph_iterator_node.py out of curiosity.  Specifically, I saw this code block under the definition for "_async_execute(self, state: dict, batchsize: int)":

#### creates a deepcopy of the graph instance for each endpoint
`
for url in urls: 
    instance = copy.copy(graph_instance)
    instance.source = url

    participants.append(instance)
`
    

The comment claims to make a deep copy of each graph instance per url, but the documentation for the copy.copy method says it only makes shallow copies unlike copy.deepcopy. I'm posting this here since I haven't experienced any explicit issues from this, but   I figured this was still worth mentioning.

**Comment by itsperini** [comment]
Hey @RyanGNguyen thanks! We switched to shallow copies because it was eventually copying rlock objects causing problems. I will change the comment ;)

**Reply by RyanGNguyen** [reply]
Ok, thanks for the clarification!

**Comment by VinciGit00** [comment]
please update to the new version
